ui/src/shared/components/PostDetail.tsx

```diff
@@ -114,6 +119,27 @@ const PostDetail: React.FC<PostDetailProps> = function (props) {
     showModal(true);
   };
 
+  const editPost = () => {
+    const postType = category;
+    const postParameters: PostInterface = {
+      id: data.id,
+      title: data.title,
+      description: data.description,
+      timestamp: data.timestamp,
+      connectedCount: data.connectedCount,
+      connectedUser: data.connectedUser,
+      active: data.active,
+      favourite: data.favourite,
+      created_by: data.created_by,
+      job_link: data.job_link,
+      company_link: data.company_link,
+    };
+
+    postContext.setPost(postParameters);
+    postContext.setType(postType);
```

Comment:
Any reason you need. to create another `postParameters` and `postType` variables?
Why not just:
```
postContext.setPost(data);
postContext.setType(category);
```

Reply:
Think initially I was facing some issue with using data and category directly, will remove and use them